backend/testing/tester.py

```diff
@@ -354,10 +354,12 @@ def test(self):
             Helper.hijack_multiple_action(
                 db_con, connection, ["i"], "hijack_action_ignore"
             )
-
             Helper.hijack_delete(
                 db_con, connection, "j", "139.5.237.0/24", "S|0|-", 136334
             )
+            Helper.load_as_sets(connection)
+
+            time.sleep(5)
```

Comment:
this is because we are doing a request right? If yes, then this is scary to use for testing as the behaviour might not be predictable (internet, etc)

Reply:
also, this makes me not able to run the test if I am offline. I think we can run tests now offline right?

Reply:
Yes we can run them offline (maybe; I have problems when running locally regarding failing RMQ channels and could not have them running besides the semaphore automation framework). How could we test the AS-SET resolution if not online though? I can remove the test but it is a part of the code that needs to be covered. I think it is good to assume online connectivity since we will also have to test RIPE, etc. monitors at some point later.

Reply:
if we use the cache in order to avoid doing multiple requests for the same AS then we can warm-up the cache from the test suite. This way we can do the request on the testing side and either warm-up the cache from the response or if we don't have connection warm-up from a pre-defined response that we have. what do you think?

Reply:
I do not get how the testing suite will warm up the cache. Note that the cache is only valid in the context of a single configuration round (this is a design decision, since when someone presses "Load AS-SETs" we should retrieve them online, as it means that a change was triggered from the user on the remote repo (or a new set was added to ARTEMIS conf). I agree that we need some backup in case the remote API does not resolve the request correctly. Do you have ideas on how to implement this? Should we check for errors and if they are present, to call load_as_sets ourselves? It seems that it will require some more complex implementation though (maybe we can do this in a separate issue).

One simple fix would be probably to have a backup config.yaml file ready to substitute the other one where the AS_SET resolution is not required at all (that means, no RIPE_WHOIS_AS_SET_... asn names are present). What is your opinion on this?

Reply:
@slowr I changed the logic in the tester. If no Internet, and with the new 10sec timeout value, the request will simply fail and no file will be written. However, for the workflow to proceed, it will use a configuration (config3.yaml) that allows the test to run correctly. If you agree with this approach let me know by resolving.